src/Angor.Test/WalletOperationsTest.cs

```diff
@@ -183,10 +202,249 @@ public void AddInputsAndSignTransaction()
+        string scriptHex = GenerateScriptHex(address, network);
+        var sendInfo = new SendInfo
+        {
+            SendToAddress = "tb1qw4vvm955kq5vrnx48m3x6kq8rlpgcauzzx63sr",
+            ChangeAddress = "tb1qw4vvm955kq5vrnx48m3x6kq8rlpgcauzzx63sr",
+            SendAmount = 0.01m,
+            SendUtxos = new Dictionary<string, UtxoDataWithPath>
+        {
+            {
+                "key", new UtxoDataWithPath
+                {
+                    UtxoData = new UtxoData
+                    {
+                        value = 1500000, // Sufficient to cover the send amount and estimated fees
+                        address = address,
+                        scriptHex = scriptHex,
+                        outpoint = new Outpoint("0000000000000000000000000000000000000000000000000000000000000000", 0),
+                        blockIndex = 1,
+                        PendingSpent = false
+                    },
+                    HdPath = "m/0/0"
+                }
+            }
+        }
+        };
+
+        // Act
+        var operationResult = await walletOperations.SendAmountToAddress(words, sendInfo);
+
+        // Assert
+        Assert.True(operationResult.Success);
+        Assert.NotNull(operationResult.Data); // ensure data is saved
+    }
+
+
+    [Fact]
+    public void GetUnspentOutputsForTransaction_ReturnsCorrectOutputs()
+    {
+        // Arrange
+        var mockHdOperations = new Mock<IHdOperations>();
+        var walletWords = new WalletWords { Words = "suspect lesson reduce catalog melt lucky decade harvest plastic output hello panel", Passphrase = "" };
+        var utxos = new List<UtxoDataWithPath>
+    {
+        new UtxoDataWithPath
+        {
+            UtxoData = new UtxoData
+            {
+                value = 1500000,
+                address = "tb1qeu7wvxjg7ft4fzngsdxmv0pphdux2uthq4z679",
+                scriptHex = "0014b7d165bb8b25f567f05c57d3b484159582ac2827",
+                outpoint = new Outpoint("0000000000000000000000000000000000000000000000000000000000000000", 0),
+                blockIndex = 1,
+                PendingSpent = false
+            },
+            HdPath = "m/0/0"
+        }
+    };
+
+        var expectedExtKey = new ExtKey();
+        mockHdOperations.Setup(x => x.GetExtendedKey(walletWords.Words, walletWords.Passphrase)).Returns(expectedExtKey);
+
+        var walletOperations = new WalletOperations(null, mockHdOperations.Object, null, null);
+
+        // Act
+        var (coins, keys) = walletOperations.GetUnspentOutputsForTransaction(walletWords, utxos);
+
+        // Assert
+        Assert.Single(coins);
+        Assert.Single(keys);
+        Assert.Equal((uint)0, coins[0].Outpoint.N);
+        Assert.Equal(1500000, coins[0].Amount.Satoshi);
+        Assert.Equal(expectedExtKey.Derive(new KeyPath("m/0/0")).PrivateKey, keys[0]);
+    }
+
+    [Fact]
+    public void CalculateTransactionFee_WithMultipleScenarios()
```

Comment:
@dangershony glad if you could go over the logic of this test,
mostly on the values, and if there is no need for Assert.Equal(0.00000001m, calculatedFeeSufficient);